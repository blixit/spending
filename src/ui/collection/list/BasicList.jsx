import { useCallback } from 'react';
import PropTypes from 'prop-types';

const BasicList = props => {
  const { items, children: renderItem } = props;
  
  const renderedItems = useCallback(() => {
    return (items || []).map(
      item => renderItem(item, item.id)
    );
  }, [items, renderItem]);

  return renderedItems();
};

BasicList.displayName = 'BasicList';

BasicList.propsType = {
  items: PropTypes.array
};

export default BasicList;
