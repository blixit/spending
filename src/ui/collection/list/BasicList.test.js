import React from 'react';

import render from 'testing/render';

import BasicList from './BasicList';

describe('[collection/list] BasicList', () => {
  const size = 5;
  const items = (new Array(size)).fill(0).map((x, id) => ({ id: id * 2 }));

  const renderList = () => (
    <div>
      <BasicList items={items} >
        {(item, id) => <span key={id}>Test {id}</span>}
      </BasicList>
    </div>
  );

  it('renders properly', () => {
    const tree = render(renderList());
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders the right number of items', () => {
    const tree = render(renderList());
    const spans = tree.root.findAllByType('span');
    expect(spans).toHaveLength(size);
  })
});
