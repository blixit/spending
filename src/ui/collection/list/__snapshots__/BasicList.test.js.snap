// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[collection/list] BasicList renders properly 1`] = `
<div>
  <span>
    Test 
    0
  </span>
  <span>
    Test 
    2
  </span>
  <span>
    Test 
    4
  </span>
  <span>
    Test 
    6
  </span>
  <span>
    Test 
    8
  </span>
</div>
`;
